aspnet/mvc/views/overview.rst

```diff
@@ -0,0 +1,222 @@
+
+``ViewData`` is a dictionary object accessed through ``string`` keys. You can store and retrieve objects in it, and you'll need to cast them to a specific type when you extract them. You can use ``ViewData`` to pass data from a controller to views, as well as within views (and partial views and layouts). String data can be stored and used directly, without the need for a cast.
+
+Set some values for ``ViewData`` in an action:
+
+.. code-block:: c#
+
+    public IActionResult SomeAction()
+    {
+        ViewData["Greeting"] = "Hello";
+        ViewData["Address"]  = new Address()
+        {
+            Name = "Steve",
+            Street = "123 Main St",
+            City = "Hudson",
+            State = "OH",
+            PostalCode = "44236"
+        };
+        
+        return View();
+    }
+
+Work with the data in a view:
+
+.. code-block:: html
+    :emphasize-lines: 3,6
+   
+    @{
+        // Requires cast
+        var address = ViewData["Address"] as Address;
+    }
+
+    @ViewData["Greeting"] World!
+
+    <address>
+        @address.Name<br />
+        @address.Street<br />
+        @address.City, @address.State @address.PostalCode
+    </address>
+
+The ``ViewBag`` objects provides dynamic access to the objects stored in ``ViewData``. This can be more convenient to work with, since it doesn't require casting. The same example as above, using ``ViewBag`` instead of a strongly typed ``address`` instance in the view:
+
+.. code-block:: html
+    :emphasize-lines: 1,4-6
+
+    @ViewBag.Greeting World!
+
+    <address>
+        @ViewBag.Address.Name<br />
+        @ViewBag.Address.Street<br />
+        @ViewBag.Address.City, @ViewBag.Address.State @ViewBag.Address.PostalCode
+    </address>
+
+.. note:: Since both refer to the same underlying ``ViewData`` collection, you can mix and match between ``ViewData`` and ``ViewBag`` when reading and writing values, if convenient.
+
+Dynamic Views
+^^^^^^^^^^^^^
+
+Views that do not declare a model type but have a model instance passed to them can reference this instance dynamically. For example, if an instance of ``Address`` is passed to a view that doesn't declare an ``@model``, the view would still be able to refer to the instance's properties dynamically as shown:
+
+.. code-block:: html
+    :emphasize-lines: 13,16-18
+    
+    <address>
+        @Model.Street<br />
+        @Model.City, @Model.State @Model.PostalCode<br />
+        <abbr title="Phone">P:</abbr>
+        425.555.0100
+    </address>
+
+This feature can offer some flexibility, but does not offer any compilation protection or intellisense. If the property doesn't exist, the page will fail at run time.
```

Comment:
Sorry I arrived late to the party:

* `Intellisense` should be `IntelliSense`
* `run time` should be `runtime`